### ISSUE TITLE
appsync: `appsync.HttpDataSourceProps` erroneously extends `BaseDataSourceProps`

### ISSUE THREAD

**Issue by ScottRobinson03** [aws/aws-cdk#31979]
### Describe the bug

In #11185, the `HttpDataSource` class was updated to extend `BackedDataSource` instead of `BaseDataSource`, however the `HttpDataSourceProps` type wasn't updated to reflect this change.

### Regression Issue

- [ ] Select this option if this issue appears to be a regression.

### Expected Behavior

The `HttpDataSourceProps` type should extend `BackendDataSourceProps`

### Current Behavior

The `HttpDataSourceProps` type erroneously extends `BaseDataSourceProps`

### Reproduction Steps

Minimum-repro to see the type error:
```ts
import * as appsync from "aws-cdk-lib/aws-appsync";
import type { IRole } from "aws-cdk-lib/aws-iam";

declare const myApi: appsync.GraphqlApi;
declare const serviceRole: IRole;

const stepFunctionHttpDataSource = new appsync.HttpDataSource(
    myApi,
    "MyStepFunctionHTTPDataSource",
    {
        api: myApi,
        authorizationConfig: {
            signingRegion: "eu-west-1",
            signingServiceName: "states",
        },
        endpoint: `https://states.eu-west-1.amazonaws.com`,
        name: "StepFunctionHTTPDataSource",
        // Providing the `serviceRole` incorrectly throws a type error
        serviceRole,
    }
);
```
![image](https://github.com/user-attachments/assets/9b6bd84d-7ccf-4324-8aae-d57a74dc00a8)

### Possible Solution

Update the `HttpDataSourceProps` type to extend `BackendDataSourceProps`

### Additional Information/Context

The only difference between `BaseDataSourceProps` and `BackedDataSourceProps`, is that `BackedDataSourceProps` adds an optional prop `serviceRole`, which is required for workflows that require custom permissions, such as the above example of starting a state machine.

The code for `BaseDataSourceProps` and `BackedDataSourceProps` is [here](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-appsync/lib/data-source.ts#L41-L48), and shown below:
```ts
/**
 * Base properties for an AppSync datasource
 */
export interface BaseDataSourceProps {
  /**
   * The API to attach this data source to
   */
  readonly api: IGraphqlApi;
  /**
   * The name of the data source
   *
   * @default - id of data source
   */
  readonly name?: string;
  /**
   * the description of the data source
   *
   * @default - None
   */
  readonly description?: string;
}

/**
 * properties for an AppSync datasource backed by a resource
 */
export interface BackedDataSourceProps extends BaseDataSourceProps {
  /**
   * The IAM service role to be assumed by AppSync to interact with the data source
   *
   * @default -  Create a new role
   */
  readonly serviceRole?: IRole;
}
```

### CDK CLI Version

2.139.0

### Node.js Version

20

### OS

MacOS Sonoma 14.7

### Language

TypeScript

### Language Version

TypeScript 5.6.3

### Other information

NB: #29689 would have fixed this however the PR was closed after going stale when requesting an exemption to requiring tests.

**Comment by khushail** [comment]
@ScottRobinson03 , thanks for opening the issue and providing the code links. 
Yes, this observation totally makes sense and I can confirm that the error seen while trying to declare the `serviceRole` -

<img width="886" alt="Screenshot 2024-11-05 at 6 51 02 PM" src="https://github.com/user-attachments/assets/2a96f2ab-383e-43aa-9f2c-1de890310525">

As its already pointed out, HttpDataSourceProps extends BaseDataSourceProps

https://github.com/aws/aws-cdk/blob/a21a0b1392faecbb0a19a6ca90b64a1da8ad5d81/packages/aws-cdk-lib/aws-appsync/lib/data-source.ts#L258

and BaseDataSourceProps has these attrtibutes-

https://github.com/aws/aws-cdk/blob/a21a0b1392faecbb0a19a6ca90b64a1da8ad5d81/packages/aws-cdk-lib/aws-appsync/lib/data-source.ts#L19C1-L36C2

```ts
export interface BaseDataSourceProps {
  /**
   * The API to attach this data source to
   */
  readonly api: IGraphqlApi;
  /**
   * The name of the data source
   *
   * @default - id of data source
   */
  readonly name?: string;
  /**
   * the description of the data source
   *
   * @default - None
   */
  readonly description?: string;
}
```

while the `serviceRole` is provided by BackedDataSourceProps-

https://github.com/aws/aws-cdk/blob/a21a0b1392faecbb0a19a6ca90b64a1da8ad5d81/packages/aws-cdk-lib/aws-appsync/lib/data-source.ts#L41

```ts
export interface BackedDataSourceProps extends BaseDataSourceProps {
  /**
   * The IAM service role to be assumed by AppSync to interact with the data source
   *
   * @default -  Create a new role
   */
  readonly serviceRole?: IRole;
}
```

which is the reason why the error is being seen.

**Comment by ScottRobinson03** [comment]
Thanks for looking into this @khushail.

What's the next step to get this fixed? Can #29689 be re-opened, or would I need to create a new PR?

**Comment by khushail** [comment]
@ScottRobinson03 , I think you should be able to reopen the PR. Also please link this issue in the description and update whatever is needed to get the community review done.

Let me know if you need any further help! Appreciate your contribution.

**Comment by ScottRobinson03** [comment]
@khushail It appears I don't have permissions to reopen the PR -- I left a comment, but that didn't reopened it. Are you perhaps able to do this on my behalf?

**Comment by khushail** [comment]
@ScottRobinson03 , its reopened. Please feel free to work on it now.

**Comment by ScottRobinson03** [comment]
Thanks @khushail. Afaik there's no more work to do from me (the PR is just a one-line fix), so it's just pending reviews (and an integ-test exemption approval, since it's just a type fix [there's no runtime code to actually test]).

**Comment by ScottRobinson03** [comment]
#29689 got automatically closed again, so I've replaced it with #32065.
